fprettify/__init__.py

```diff
@@ -174,6 +175,27 @@
 
 ENDANY_RE = re.compile(SOL_STR + r"END" + EOL_STR, RE_FLAGS)
 
+# Regular expressions for preprocessor directives
+
+PREPRO_DEF_RE = re.compile(r"#:DEF\s+.*\n", RE_FLAGS)
+PREPRO_ENDDEF_RE = re.compile(r"#:ENDDEF.*\n", RE_FLAGS)
+
+PREPRO_IF_RE = re.compile(r"#:IF\s+.*\n", RE_FLAGS)
+PREPRO_ELIF_ELSE_RE = re.compile(r"#:(ELIF\s+.*\n|ELSE\n)", RE_FLAGS)
+PREPRO_ENDIF_RE = re.compile(r"#:ENDIF\s*\n", RE_FLAGS)
+
+PREPRO_FOR_RE = re.compile(r"#:FOR\s+.*\n", RE_FLAGS)
+PREPRO_ENDFOR_RE = re.compile(r"#:ENDFOR\s*\n", RE_FLAGS)
+
+PREPRO_BLOCK_RE = re.compile(r"#:BLOCK\s+.*\n", RE_FLAGS)
+PREPRO_ENDBLOCK_RE = re.compile(r"#:ENDBLOCK\s+.*\n", RE_FLAGS)
+
+PREPRO_CALL_RE = re.compile(r"#:CALL\s+.*\n", RE_FLAGS)
+PREPRO_ENDCALL_RE = re.compile(r"#:ENDCALL\s+.*\n", RE_FLAGS)
+
+PREPRO_MUTE_RE = re.compile(r"#:MUTE\s*\n", RE_FLAGS)
+PREPRO_ENDMUTE_RE = re.compile(r"#:ENDMUTE\s*\n", RE_FLAGS)
+
```

Comment:
Why do you want to parse fypp directives? Do you want to indent fypp directives based on fypp syntax? Or do you want to adjust the indentation of fypp directives based on the Fortran syntax?

Reply:
Hello, 

I only added fypp preprocessor directives (#:) because, it is needed for the DFTB+ formatting. The other fypp directives, could be added later, but for the moment they are not needed for DFTB+. 

We would like to adapt the indentation of the fypp preprocessor directives to the Fortran syntax. So if a preprocessor directive comes in a block construct, it should be indented normally.

Reply:
Ok I see. Then the regular expressions I commented above are not needed, right? In this case, could you remove them?

Reply:
Hello, 

I've removed the regular expressions that were not needed, and the lists that included them. They were just remants of something I wanted to add before.